From 683add082c1da84fd56d1cd05e0b2989cf8bdde5 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Fri, 26 Jan 2018 11:27:35 +0000
Subject: [PATCH 1/3] msm8974-common: Binderize them all

 * Switch to binderized HAL services as possible and update
   HIDL manifest accordingly.

 * Following considerations regarding those without any change:

   - Sensors left out of the game because wouldn't work
     otherwise.

Change-Id: Id50291488d655187aa013c51bdd6890dca010564
---
 manifest.xml                             | 12 ++++++------
 msm8974.mk                               |  7 ++++++-
 sepolicy/common/hal_bluetooth_default.te |  7 +++++++
 sepolicy/common/hal_gnss_default.te      |  3 +++
 sepolicy/common/hal_keymaster_default.te |  1 +
 sepolicy/common/hal_memtrack_default.te  |  1 +
 6 files changed, 24 insertions(+), 7 deletions(-)
 create mode 100644 sepolicy/common/hal_bluetooth_default.te
 create mode 100644 sepolicy/common/hal_gnss_default.te
 create mode 100644 sepolicy/common/hal_keymaster_default.te
 create mode 100644 sepolicy/common/hal_memtrack_default.te

diff --git a/manifest.xml b/manifest.xml
index f7346d2..e735240 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -1,7 +1,7 @@
 <manifest version="1.0" type="device">
     <hal format="hidl">
         <name>android.hardware.audio</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>2.0</version>
         <interface>
             <name>IDevicesFactory</name>
@@ -10,7 +10,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.audio.effect</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>2.0</version>
         <interface>
             <name>IEffectsFactory</name>
@@ -19,7 +19,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.bluetooth</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>1.0</version>
         <interface>
             <name>IBluetoothHci</name>
@@ -52,7 +52,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.gnss</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>1.0</version>
         <interface>
             <name>IGnss</name>
@@ -88,7 +88,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.keymaster</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>3.0</version>
         <interface>
             <name>IKeymasterDevice</name>
@@ -97,7 +97,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.memtrack</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>1.0</version>
         <interface>
             <name>IMemtrack</name>
diff --git a/msm8974.mk b/msm8974.mk
index eb470ee..bbfd3de 100644
--- a/msm8974.mk
+++ b/msm8974.mk
@@ -65,6 +65,7 @@ PRODUCT_PACKAGES += \
 # Audio
 PRODUCT_PACKAGES += \
     android.hardware.audio@2.0-impl \
+    android.hardware.audio@2.0-service \
     android.hardware.audio.effect@2.0-impl \
     audio_policy.msm8974 \
     audio.a2dp.default \
@@ -80,6 +81,7 @@ PRODUCT_PACKAGES += \
 # Bluetooth
 PRODUCT_PACKAGES += \
     android.hardware.bluetooth@1.0-impl \
+    android.hardware.bluetooth@1.0-service \
     libbt-vendor
 
 # Boot animation
@@ -91,6 +93,7 @@ PRODUCT_PACKAGES += \
     android.hardware.graphics.allocator@2.0-service \
     android.hardware.graphics.mapper@2.0-impl \
     android.hardware.memtrack@1.0-impl \
+    android.hardware.memtrack@1.0-service \
     copybit.msm8974 \
     gralloc.msm8974 \
     hwcomposer.msm8974 \
@@ -104,6 +107,7 @@ PRODUCT_PACKAGES += \
 # GPS
 PRODUCT_PACKAGES += \
     android.hardware.gnss@1.0-impl \
+    android.hardware.gnss@1.0-service \
     gps.msm8974
 
 PRODUCT_COPY_FILES += \
@@ -125,7 +129,8 @@ PRODUCT_COPY_FILES += \
 
 # Keymaster
 PRODUCT_PACKAGES += \
-    android.hardware.keymaster@3.0-impl
+    android.hardware.keymaster@3.0-impl \
+    android.hardware.keymaster@3.0-service
 
 # Media
 PRODUCT_COPY_FILES += \
diff --git a/sepolicy/common/hal_bluetooth_default.te b/sepolicy/common/hal_bluetooth_default.te
new file mode 100644
index 0000000..2e32627
--- /dev/null
+++ b/sepolicy/common/hal_bluetooth_default.te
@@ -0,0 +1,7 @@
+allow hal_bluetooth_default bluetooth_device:chr_file w_file_perms;
+allow hal_bluetooth_default bt_fw_file:file r_file_perms;
+allow hal_bluetooth_default efs_file:dir search;
+allow hal_bluetooth_default firmware_file:dir r_dir_perms;
+allow hal_bluetooth_default proc_bt_sleep:dir w_dir_perms;
+allow hal_bluetooth_default proc_bt_sleep:file w_file_perms;
+allow hal_bluetooth_default wifi_data_file:file r_file_perms;
diff --git a/sepolicy/common/hal_gnss_default.te b/sepolicy/common/hal_gnss_default.te
new file mode 100644
index 0000000..f9aaacc
--- /dev/null
+++ b/sepolicy/common/hal_gnss_default.te
@@ -0,0 +1,3 @@
+allow hal_gnss_default location_data_file:file rw_file_perms;
+
+qmux_socket(hal_gnss_default)
diff --git a/sepolicy/common/hal_keymaster_default.te b/sepolicy/common/hal_keymaster_default.te
new file mode 100644
index 0000000..b5d94ea
--- /dev/null
+++ b/sepolicy/common/hal_keymaster_default.te
@@ -0,0 +1 @@
+r_dir_file(hal_keymaster_default, firmware_file)
diff --git a/sepolicy/common/hal_memtrack_default.te b/sepolicy/common/hal_memtrack_default.te
new file mode 100644
index 0000000..35cc326
--- /dev/null
+++ b/sepolicy/common/hal_memtrack_default.te
@@ -0,0 +1 @@
+dontaudit hal_memtrack_default { domain -surfaceflinger }:file read;
-- 
2.15.1

